File: app/src/main/java/com/left/gank/ui/gallery/GalleryPagerAdapter.java
package com.left.gank.ui.gallery;

import android.business.domain.Gift;

import java.util.List;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

public class GalleryPagerAdapter extends FragmentStatePagerAdapter {
    private List<Gift> gifts;

    GalleryPagerAdapter(FragmentManager fragmentManager) {
        super(fragmentManager);
    }

    void setGifts(List<Gift> gifts) {
        this.gifts = gifts;
    }

    @Override
    public int getCount() {
        return gifts == null ? 0 : gifts.size();
    }

    @Override
    public Fragment getItem(int position) {
        return GalleryFragment.newInstance(gifts.get(position).imgUrl);
    }
}

File: app/src/main/java/com/left/gank/ui/gallery/GalleryPagerAdapter.kt
package com.left.gank.ui.gallery

import android.business.domain.Gift

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter

class GalleryPagerAdapter internal constructor(fragmentManager: FragmentManager) : FragmentStatePagerAdapter(fragmentManager) {
    private var gifts: List<Gift>? = null
    private var url: String? = null
    internal fun setGifts(gifts: List<Gift>, url: String?) {
        this.gifts = gifts
        this.url = url
    }

    override fun getCount(): Int {
        return if (gifts == null) 0 else gifts!!.size
    }

    override fun getItem(position: Int): Fragment {
        return GalleryFragment.newInstance(gifts!![position])
    }
}